### PR TITLE
fix ebs volume mounting behavior

### PR DESCRIPTION
**Pull request by wlan0**
@alena1108 - Moved the commit to v1.3.4-rancher

### CODE REVIEW

**Comment by alena1108** [comment]
@wlan0 thanks! Have you figured out why it wasn't working before on your 1.3.4 branch?

File: pkg/cloudprovider/providers/aws/aws.go [hunk 1]
```diff
@@ -2907,13 +2906,12 @@ func (c *Cloud) findInstanceByNodeName(nodeName string) (*ec2.Instance, error) {
 	if err != nil {
 		return nil, err
 	}
-	if len(instances) == 0 {
-		return nil, nil
-	}
-	if len(instances) > 1 {
-		return nil, fmt.Errorf("multiple instances found for name: %s", nodeName)
+	for _, instance := range instances {
+		if strings.Contains(*instance.PrivateDnsName, nodeName) {
```
Comment (alena1108) [inline pkg/cloudprovider/providers/aws/aws.go]: should it be "strings.hasPrefix()" instead?

Comment (wlan0) [inline pkg/cloudprovider/providers/aws/aws.go]: both `Contains()` and `hasPrefix()` work. I can change it to `hasPrefix()`

**Comment by wlan0** [comment]
@alena1108  I changed it to `HasPrefix()` instead of `Contains()`

**Comment by wlan0** [comment]
@ibuildthecloud - I've updated the PR to start kubelet with fqdn. we can merge this after merging python-agent.

**Comment by alena1108** [comment]
This one can be probably closed.

**Pull request closed without merging**